webapp/graphite/render/functions.py

```diff
@@ -1013,18 +1013,21 @@ def weightedAverage(requestContext, seriesListAvg, seriesListWeight, *nodes):
   """
   sortedSeries={}
 
+  if len(seriesListAvg) != len(seriesListWeight):
```

Comment:
what happens if the length is the same but the `key` generated from `seriesAvg` and `seriesWeight` doesn't always match? If it's possible that `seriesListAvg` and `seriesListWeight` have different lengths, wouldn't it also be possible that they have the same length but some series not matching when keying by the `aggKey()`?

Reply:
that is a good point! I guess that will need to be another PR

Reply:
wouldn't it be easy to catch that case?

instead of assigning to `key` on `:1020` just assign to `avgKey` and on `:1026` to `weightKey` and then compare the two, if they're different that means that further down on `:1044` there's going to be an exception because it will try to access the property `.name` of `None`.

Reply:
> wouldn't it be easy to catch that case?
> 
> instead of assigning to `key` on `:1020` just assign to `avgKey` and on `:1026` to `weightKey` and then compare the two, if they're different that means that further down on `:1044` there's going to be an exception because it will try to access the property `.name` of `None`.
> (I didn't check how `izip_longest()` works)

the series are not necessarily in the same order

Reply:
Ok, ic. Then let's just merge this as it is already an improvement compared to the previous situation, even if it's not perfect yet.